spec/controllers/users/registrations_controller_spec.rb

```diff
@@ -74,8 +74,7 @@
 
         before { subject }
 
-        it { expect(response).to redirect_to(root_path) }
-        it { expect(flash.notice).to eq(I18n.t('devise.registrations.signed_up_but_unconfirmed')) }
+        it { expect(response).to redirect_to(new_user_confirmation_path(user: { email: user[:email] })) }
```

Comment:
Je pense qu'on devrait expliquer ici pourquoi on veut ce comportement. Sinon, même s'il est techniquement testé, si on n'explique pas _pourquoi_, un·e développeur·se futur risque de modifier ce comportement, sans comprendre que ça casse l'intention derrière.

Je propose un truc du genre :

```ruby
context 'and the user is confirmed' do
  # …

  it 'sends an email to the user, stating that the account already exists' do
    expect(UserMailer).to have_received(:new_account_warning)
  end

  it 'avoids leaking information about the account existence, by redirecting to the same page than normal signup' do
   expect(response).to redirect_to(new_user_confirmation_path(user: { email: user[:email] }))
  end
end

context 'and the user is not confirmed' do
  # …

  it 'does not send a warning email' do
    expect(UserMailer).not_to have_received(:new_account_warning)
  end

  it 'avoids leaking information about the account existence, by redirecting to the same page than normal signup' do
   expect(response).to redirect_to(new_user_confirmation_path(user: { email: user[:email] }))
  end
end
```

What do you think?

Reply:
@kemenaran J'ai changé le spec.